src/solver/variable/info.h

```diff
@@ -381,7 +381,15 @@ struct VariableAccessor<ResultsT, Category::dynamicColumns>
         assert(results.data.area && "Area is NULL");
         const bool thermal_details = fileLevel & Category::de;
         const bool renewable_details = fileLevel & Category::de_res;
-        if (thermal_details && renewable_details)
+        const bool st_storage_details = fileLevel & Category::de_sts;
+
+        std::array<bool, 3> kind_of_details = { thermal_details, renewable_details , st_storage_details };
```

Comment:
You could use [`std::bitset`](https://en.cppreference.com/w/cpp/utility/bitset) and [`std::bitset::count`](https://en.cppreference.com/w/cpp/utility/bitset/count) to do exactly this.

Reply:
fileLevel would have to be a `std::bitset`